client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/iteration/SegmentKeyTracker.java

```diff
@@ -17,20 +18,21 @@
  * @author gustavonalle
  * @since 8.0
  */
-public class SegmentKeyTracker implements KeyTracker {
+class SegmentKeyTracker implements KeyTracker {
 
    private static final Log log = LogFactory.getLog(SegmentKeyTracker.class);
 
    private final AtomicReferenceArray<Set<byte[]>> keysPerSegment;
    private final SegmentConsistentHash segmentConsistentHash;
 
-   public SegmentKeyTracker(SegmentConsistentHash segmentConsistentHash) {
+   public SegmentKeyTracker(SegmentConsistentHash segmentConsistentHash, Set<Integer> segments) {
       int numSegments = segmentConsistentHash.getNumSegments();
       keysPerSegment = new AtomicReferenceArray<>(numSegments);
       if (log.isDebugEnabled()) log.debugf("Created SegmentKeyTracker with %d segments", numSegments);
       this.segmentConsistentHash = segmentConsistentHash;
-      IntStream.range(0, segmentConsistentHash.getNumSegments())
-              .forEach(i -> keysPerSegment.set(i, CollectionFactory.makeSet(ByteArrayEquivalence.INSTANCE)));
+      Stream<Integer> segmentStream = segments == null ?
```

Comment:
I am thinking it would be better to have this an IntStream instead.

``Stream<Integer> case``
segments is null we have to box N values and unbox N values in the forEach consumer
segments is not null we have to unbox N values in the forEach consumer

``IntStream case``
segments is non null we have 0 unboxing
segment is not null we have 1 unboxing in a call to ``mapToInt``

In either case it should be better.